python/cog/server/http.py

```diff
@@ -239,14 +239,14 @@ async def start_shutdown() -> Any:
             shutdown_event.set()
         return JSONResponse({}, status_code=200)
 
-    def _check_setup_result() -> Any:
+    async def _check_setup_result() -> Any:
         if app.state.setup_result is None:
             return
 
-        if not app.state.setup_result.ready():
+        if not app.state.setup_result.done():
             return
 
-        result = app.state.setup_result.get()
+        result = await app.state.setup_result
```

Comment:
I think it might be clearer to do

```suggestion
        result = app.state.setup_result.result()
```

here, because as written, this kind of implies that we might actually be waiting for it, which shouldn't happen.

Reply:
Also do we need to handle cancellation of this task here?

Reply:
hm, yes, both result and await can raise CancelledError (or errors from setup). that code path would happen if we get a healthcheck while shutting down after being told to shut down... is that SETUP_FAILED? there's not a shutting-down state

Reply:
it's possible just propagating CancelledError and setup errors through to the healthcheck isn't the worst behavior, it should return 500 which seems reasonable